docs: arabic translation (#791)
t3-oss/create-t3-app: www/src/pages/ar/faq.md

---
title: الأسئلة الشائعة
description: الأسئلة المتكررة حول إنشاء تطبيق T3
layout: ../../layouts/docs.astro
lang: ar
dir: rtl
---

ها هي بعض الأسئلة التي يكثر سؤالها عن `create-t3-app`.

## ماذا بعد ؟ كيف أقوم بتوليد مشروع باستخدام `create-t3-app` ؟

نحن نحاول ان نبقي هذا المشروع بسيطاًً قدر الامكان، لذلك فقد وضعنا حجز الاساس لك، ويمكن اضافة ما تريد وقت الحاجه عندما يكون ضرورياَ.

> اذا كنت غير ملم ببعض التقنيات المستخدمة في هذا المشروع الرجاء الاتجاة الي الـ Docs المعنيه بها، ذإذا ؤاجهتك مشاكل اخري فالرجاء الانضمام الي سيرفر الديسكور الخاص بنا [Discord](https://t3.gg/discord) and ask for help

Reviewer: ```suggestion
إذا كنت غير ملم ببعض التقنيات المستخدمة في هذا المشروع الرجاء الاتجاه ألى الـ Docs المعنيه بها، إذا واجهتك مشكلات أخري فالرجاء الانضمام ألى سيرفرنا الديسكور [Discord](https://t3.gg/discord) وأطلب المساعدة
```

يمكنك حذف `خاص بنا` وأمثالها

Author: Thanks for the help man, These types happened bc I'm on the run to finish the translation Before Friday, I was planning for a revision round any way.

Would u like to join me?

Reviewer: of course 👍

Author: Join the Discord Server https://discord.gg/t3dotgg and join `ct3a-meta` channel, I'm waiting there :)

### مقالات

- [Build a full stack app with create-t3-app](https://www.nexxel.dev/blog/ct3a-guestbook)
- [A first look at create-t3-app](https://dev.to/ajcwebdev/a-first-look-at-create-t3-app-1i8f)
- [Migrating your T3 App into a Turborepo](https://www.jumr.dev/blog/t3-turbo)

### فيديوهات

- [Build a Blog With the T3 Stack - tRPC, TypeScript, Next.js, Prisma & Zod](https://www.youtube.com/watch?v=syEWlxVFUrY)
- [Build a Live Chat Application with the T3 Stack - TypeScript, Tailwind, tRPC](https://www.youtube.com/watch?v=dXRRY37MPuk)
- [The T3 Stack - How We Built It](https://www.youtube.com/watch?v=H-FXwnEjSsI)
- [An overview of the create T3 App (Next, Typescript, Tailwind, tRPC, Next-Auth)](https://www.youtube.com/watch?v=VJH8dsPtbeU)

## لماذا هناك ملفات بامتداد `.js` في هذا المشروع ؟

وفقًا لـ [T3-Axiom #3](/en/introduction#typesafety-isnt-optional) ، نحن نعتبر `Typesafety` كمواطن من الدرجة الأولى. لسوء الحظ ، لا تدعم جميع الـ Frameworks والاضافات TypeScript مما يعني أن بعض ملفات التكوين يجب أن تكون ملفات JS.

> نحن نحاول التأكيد على أن هذه الملفات هي `Javascript ` لاسباب خارجة عن ارادتنا ، من خلال التصريح صراحةً عن نوع كل ملف (cjs أو mjs) اعتمادًا على ما تدعمه المكتبة التي تستخدمها. أيضًا ، لكن لا تزال جميع ملفات js في هذا المشروع تخضع لفحص `Types `باستخدام تعليق` @ ts-check` في الأعلى.

Reviewer: ```suggestion
نحن نحاول التأكيد على أن هذه الملفات هي `Javascript ` لأسباب خارجة عن أرادتنا ، بواسطة التصريح صراحةً عن نوع كل ملف (cjs أو mjs) اعتمادًا على ما تدعمه المكتبة التي تستخدمها. أيضًا ، لكن لا تزال جميع ملفات js في هذا المشروع تخضع لفحص `Types` باستخدام تعليق `ts-check@` في الأعلى.
```

هو فيه سبب لكتابة `ts-check@`  مثل `ts-check@`, وهل يمكن هل هذا السبب لأن قراء الشاشة (screen reader) سوف يقرئه غلط, فسوف يستخدمه غلط.

## أجد صعوبة في إضافة `i18n ` إلى تطبيقي. هل هناك أي مرجع يمكنني الرجوع إلية؟

لقد قررنا عدم تضمين i18n افتراضيًا في "create-t3-app" وذلك لأنه موضوع شائك للغاية وهناك العديد من الطرق لتنفيذة.
ومع ذلك ، إذا كنان لابُد من تَنفيذه وترغب في رؤية مشروع مَرجعي ، فلدينا مَرجع  يُوضح كيف يمكنك إضافة i18n إلى تَطبيق T3 باستخدام next-i18next).

> ## لماذا نستخدم `pages/` وليس `app/` في Next.js 13 ؟

Reviewer: ```suggestion
## لماذا نستخدم `pages/` وليس `app/` في Next.js 13؟

> كما ذكرنا مُسبقا في [T3-Axiom #2](/en/introduction#bleed-responsibly)، نحن نُحب الأشياء المتطورة لكننا نُقدر الاستقرار ، فمن الصعب تَحويل الـ router الي النظام الجديد، شاهد [not a great place to bleed](https://youtu.be/mnwUbtieOuI?t=1662).
> فـ `/app` هو مُجرد لَمحة عن المستقبل ، فهو ليس جاهزًا للإنتاج ؛ غير ان الـ (API) في مَرحلة تجريبية ومن المتوقع أن تحدث له تغييرات جذرية.

Reviewer: ```suggestion
كما ذكرنا مُسبقا في [T3-Axiom #2](/en/introduction#bleed-responsibly)، نحن نُحب الأشياء المتطورة لكننا نُقدر الاستقرار ، فمن الصعب تَحويل الـ router ألى النظام الجديد، شاهد [not a great place to bleed](https://youtu.be/mnwUbtieOuI?t=1662).
فأن `/app` مُجرد لَمحة ألى المستقبل، فهو ليس جاهزًا للإنتاج؛ غير أن الـ (API) في مَرحلة تجريبية ومن المتوقع أن تحدث له تغييرات جذرية.
```

> لمعرفة الميزات المدعومة والمخطط العمل عليها في / app dir ، قم بزيارة [beta Next.js docs](https://beta.nextjs.org/docs/app-directory-roadmap#supported-and-planned-features)

Reviewer: ```suggestion
لمعرفة الميزات المدعومة والمخطط العمل عليها في `/app`، أقرأ [beta Next.js docs](https://beta.nextjs.org/docs/app-directory-roadmap#supported-and-planned-features)
```
